tests/rptest/services/action_injector.py

```diff
@@ -0,0 +1,183 @@
+import dataclasses
+import random
+import time
+from threading import Thread
+from typing import Optional, List
+
+from ducktape.cluster.cluster import ClusterNode
+from ducktape.utils.util import wait_until
+
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.services.failure_injector import FailureSpec, FailureInjector
+from rptest.services.redpanda import RedpandaService
+
+
+@dataclasses.dataclass
+class ActionConfig:
+    cluster_start_lead_time_sec: float
+    min_time_between_actions_sec: float
+    max_time_between_actions_sec: float
+    max_affected_nodes: Optional[int] = None
+
+    def random_time_in_range(self) -> float:
+        return random.uniform(self.min_time_between_actions_sec, self.max_time_between_actions_sec)
+
+
+class RandomNodeOp:
+    def __init__(self, redpanda: RedpandaService, config: ActionConfig, admin: Admin):
+        self.admin = admin
+        self.config = config
+        self.redpanda = redpanda
+        self.nodes_affected = set()
+
+    def limit_reached(self) -> bool:
+        raise NotImplementedError
+
+    def target_node(self) -> ClusterNode:
+        available = set(self.redpanda.nodes) - self.nodes_affected
+        if available:
+            return random.choice(list(available))
+
+    def action(self):
+        raise NotImplementedError
+
+    def __call__(self, *args, **kwargs):
+        if not self.limit_reached():
+            self.action()
+
+
+class RandomNodeDecommission(RandomNodeOp):
+    def limit_reached(self):
+        return len(self.nodes_affected) >= self.config.max_affected_nodes
```

Comment:
we have `nodes_affected` and `max_affected_nodes` let's rename the former to affected_nodes

Reply:
fixed